Tests/Client/Custom Assertions/AssertJSONEqual.swift

```diff
@@ -25,34 +25,52 @@ func AssertJSONEqual(_ expression1: @autoclosure () throws -> Data,
                      _ expression2: @autoclosure () throws -> Data,
                      file: StaticString = #file,
                      line: UInt = #line) {
+    
+    guard let error = try CheckJSONEqual(expression1(), expression2()) else { return }
+    XCTFail("Error: \(error)", file: file, line: line)
+}
+
+/// Compares the given 2 JSON Serializations are equal, by creating JSON objects from Data and comparing dictionaries.
+/// Recursively calls itself for nested dictionaries and returns an error if the expressions are not equal.
+/// - Parameters:
+///   - expression1: JSON object 1, as Data. From string, you can do `Data(jsonString.utf8)`
+///   - expression2: JSON object 2, as Data.
+func CheckJSONEqual(_ expression1: @autoclosure () throws -> Data,
```

Comment:
These changes should address this commet: https://github.com/GetStream/stream-chat-swift/pull/364#discussion_r472039410